gusto/physics.py

```diff
@@ -677,6 +684,22 @@ def __init__(self, equation, saturation_curve, vapour_name="water_vapour",
         Vv = W.sub(self.Vv_idx)
         test_v = equation.tests[self.Vv_idx]
 
+        # Check if saturation is a function, and if so what of.
+        # Currently this works only when saturation is depth-dependent or not
+        # changing in time.
+        # TODO: allow saturation to be dependent on other variables
+        if isinstance(self.saturation, FunctionType):
+            if self.saturation_variable == "depth":
+                self.variable_idx = equation.field_names.index("D")
+                Vvar_space = W.sub(self.variable_idx)
+                self.variable = Function(Vvar_space)
+                self.sat_func = Function(Vvar_space)
```

Comment:
would prefer not to abbreviate too much - so `saturation_function` instead of `sat_func`